pkg/proc/disasm_arm64.go

```diff
@@ -0,0 +1,247 @@
+package proc
+
+import (
+	"fmt"
+	"io"
+
+	"golang.org/x/arch/arm64/arm64asm"
+)
+
+var maxInstructionLength uint64 = 4
+
+type archInst arm64asm.Inst
+
+//type archInst x86asm.Inst
+
+func asmDecode(mem []byte, pc uint64) (*archInst, error) {
+	inst, err := arm64asm.Decode(mem)
+	if err != nil {
+		return nil, err
+	}
+	patchPCRel(pc, &inst)
+	r := archInst(inst)
+	return &r, nil
+}
+
+func (inst *archInst) Size() int {
+	return 4
+}
+
+// converts PC relative arguments to absolute addresses
+func patchPCRel(pc uint64, inst *arm64asm.Inst) {
+	//var Imm64 arm64asm.Imm64
+
+	for i := range inst.Args {
+		rel, isrel := inst.Args[i].(arm64asm.PCRel)
+		if isrel {
+
+			inst.Args[i] = arm64asm.ArchArm64(uint64(pc) + uint64(rel) + uint64(inst.Len))
+
+		}
+	}
+}
+
+// Text will return the assembly instructions in human readable format according to
+// the flavour specified.
+var readerAt io.ReaderAt
```

Comment:
Remove this, pass nil to arm64asm.GoSyntax instead. If I'm not mistaken this variable is always nil nothing reads or writes it and only exists because arm64asm.GoSyntax has it as an argument (which it doesn't use). It makes Text look non-reentrant.